process_preds.Rmd

```diff
@@ -59,15 +57,15 @@ process_file <- function(input_path, template_path, resolution, save = FALSE, ou
     # Set CRS to match the shapefile
     crs(rast_shapefile) <- st_crs(shapefile)$proj4string
 
-    # Extract filename without extension for naming
-    file_name <- tools::file_path_sans_ext(basename(input_path))
+    # Extract filename without extension for naming 
+    file_name <- tools::file_path_sans_ext(basename(predictor_path))
 
-    # Save as an independent object
-    assign(file_name, rast_shapefile, envir = .GlobalEnv)
+    # Store the processed object with the filename
+    output_object <- list(name = file_name, object = rast_shapefile)
```

Comment:
I would just change this to `output_object <- rast_shapefile` since we want the returned object to be a raster instead of a list of a raster.